### PR TITLE
Better error handling for email

### PR DESCRIPTION
**Pull request by mjoyce91**
- Better error handling for when a client is not a TalentMAP user and we try to email them
- Email the bidder whose handshake has been registered

### CODE REVIEW

File: talentmap_api/common/common_helpers.py [hunk 1]
```diff
@@ -538,11 +538,18 @@ def registered_handshake_notification_thread(cp_id, jwt, perdet_to_exclude, is_a
     results = get_bureau_position_bids(cp_id, {}, jwt, '')
     emailAddresses = pydash.reject(results, lambda x : pydash.to_string(x.get('emp_id')) == pydash.to_string(perdet_to_exclude))
     emailAddresses = pydash.map_(emailAddresses, 'email')
+    bidderAddress = pydash.filter_(results, lambda x : pydash.to_string(x.get('emp_id')) == pydash.to_string(perdet_to_exclude))
+    bidderAddress = pydash.map_(bidderAddress, 'email')
```
Comment (elizabeth-jimenez) [inline talentmap_api/common/common_helpers.py:542]: `bidderAddress = pydash.pluck(pydash.filter_(results, lambda x : pydash.to_string(x.get('emp_id')) == pydash.to_string(perdet_to_exclude)), 'email')` 😈 

i kid!

File: talentmap_api/common/common_helpers.py [hunk 2]
```diff
@@ -538,11 +538,18 @@ def registered_handshake_notification_thread(cp_id, jwt, perdet_to_exclude, is_a
...
     def send_message (email):
         if email:
             message = f"Another bidder's handshake has been {action} for a position that you bid on."
             send_email(message, message, [email])
+    def send_message_to_registered (email):
+        if email:
+            message = f"Your handshake for a position that you bid on has been {action} by a CDO."
+            send_email(message, message, [email])
     pydash.for_each(emailAddresses, send_message)
+    pydash.for_each(bidderAddress, send_message_to_registered)
```
Comment (elizabeth-jimenez) [inline talentmap_api/common/common_helpers.py:552]: this should always be a single email, right?

Comment (mjoyce91) [reply talentmap_api/common/common_helpers.py:552]: yes

File: talentmap_api/common/common_helpers.py [hunk 3]
```diff
@@ -538,11 +538,18 @@ def registered_handshake_notification_thread(cp_id, jwt, perdet_to_exclude, is_a
     results = get_bureau_position_bids(cp_id, {}, jwt, '')
     emailAddresses = pydash.reject(results, lambda x : pydash.to_string(x.get('emp_id')) == pydash.to_string(perdet_to_exclude))
     emailAddresses = pydash.map_(emailAddresses, 'email')
```
Comment (elizabeth-jimenez) [inline talentmap_api/common/common_helpers.py:540]: we can save a `filter` and do `bidderAddress = pydash.difference(results, emailAddresses)` instead?

File: talentmap_api/common/common_helpers.py (hunk 2 above)
Comment (elizabeth-jimenez) [reply talentmap_api/common/common_helpers.py:552]: ^discussed in slack

**Review by elizabeth-jimenez — APPROVED** [review]
🍰 👍

File: talentmap_api/common/common_helpers.py (hunk 3 above)
Comment (mjoyce91) [reply talentmap_api/common/common_helpers.py:540]: I think I prefer the logic just being reversed (`filter` instead of `reject`)

**Pull request merged**